### ISSUE TITLE
SetTimeContext with null arguments still crashes

### ISSUE THREAD

**Issue by GabrieleManduchi** [MDSplus/mdsplus#1240]
There are still seriuous problems with SetTimeContext that prevent the usage of the current alpha branch. When called directly SetTimeContext with null arguments, it works. When called from mdsip, it crashes unless a tree has been previously open.

To replicate the issue:
1) Launch mdsip on a server node
2) On the client: tdic
tdic> mdsconnect(<server>)
tdic>mdsvalue('SetTimeContext(*,*,*)')   ---------> mdsip on server crashes!!

Note that if a tree has been open previously (tdic> mdsopen(<tree>, shot) , SetTimeContext() works

**Comment by GabrieleManduchi** [comment]
Some more information: the problem is the usual one: if no tree has been open before, the mdsip program crashes because timecontext tc is NULL (see below) and a segev is generated at line  INIT_STATUS_AS MdsCopyDxXd(start, &tc->start);


 /*****************************************
  TimeContext sticks with current db (tree)
  *****************************************/
 int _TreeSetTimeContext(void *dbid, struct descriptor *start, struct descriptor *end, struct descriptor *delta){
   timecontext_t* tc = &((PINO_DATABASE*)dbid)->timecontext;
   INIT_STATUS_AS MdsCopyDxXd(start, &tc->start);
   if STATUS_OK {
     status = MdsCopyDxXd(end, &tc->end);
     if STATUS_OK
       status = MdsCopyDxXd(delta, &tc->delta);
   }
   return status;
 }

**Comment by zack-vii** [comment]
Hi @GabrieleManduchi: Could you please verify the version of the mdsip service. i just tried it
```tdi
mdsconnect('localhost:4000')
1
TDI> mdsvalue('SetTimeContext()')
1
TDI> mdsvalue('SetTimeContext(,,*)')
1
```

btw: how would you do that without hack via system calls. we should implement a version() method in tdi/tcl

PS: try
```tdi
write(*,mdsvalue("tcl('SHOW VERSION',_o);_o"))
```

**Comment by zack-vii** [comment]
```
~$ tdic
TDI> mdsconnect('mds-data-2:8000')
1
TDI> mdsvalue('SetTimeContext(,,*)')
1
TDI> write(*,mdsvalue("tcl('SHOW VERSION',_o);_o"))


MDSplus version: 7.30.3
Git branch: alpha
  Git tag: alpha_release-7-30-3
  View:
    https://github.com/MDSplus/mdsplus/tree/alpha_release-7-30-3
  Download:
    https://github.com/MDSplus/mdsplus/archive/alpha_release-7-30-3.tar.gz
  Git commit: b62fa8bf0c01139106301fb1700d2db7752d4aa1
  Build date: Wed Feb 14 03:16:26 UTC 2018



337
```

**Comment by zack-vii** [comment]
should be fixed by pr #1243

**Comment by GabrieleManduchi** [comment]
Ok, seems to work, I close the issue
